From 4cd23bdde90cc7d2911106017c60617e96d9d1f4 Mon Sep 17 00:00:00 2001
From: jamal <hadi@cyberus.ca>
Date: Fri, 8 Aug 2008 10:06:17 -0400
Subject: [PATCH 01/23] ip: Allow for easier debug of buggy devices that dont send their names

patch attached this time..

On Fri, 2008-08-08 at 10:01 -0400, jamal wrote:
> wireless drivers using wext is a prime example if you need a test case.
>
> cheers,
> jamal

ip: Allow for easier debug of buggy devices that dont send their names

With the old message couldnt tell which device had the bug.
This patch provides at least an ifindex to narrow it down.
Theres also no point in bailing out because of one bug; we
allow it to go on so we could dump as much info as we can

Signed-off-by: Jamal Hadi Salim <hadi@cyberus.ca>
---
 ip/ipaddress.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/ip/ipaddress.c b/ip/ipaddress.c
index 28d3c2d..b055b68 100644
--- a/ip/ipaddress.c
+++ b/ip/ipaddress.c
@@ -210,8 +210,7 @@ int print_linkinfo(const struct sockaddr_nl *who,
 
 	parse_rtattr(tb, IFLA_MAX, IFLA_RTA(ifi), len);
 	if (tb[IFLA_IFNAME] == NULL) {
-		fprintf(stderr, "BUG: nil ifname\n");
-		return -1;
+		fprintf(stderr, "BUG: device with ifindex %d has nil ifname\n", ifi->ifi_index);
 	}
 	if (filter.label &&
 	    (!filter.family || filter.family == AF_PACKET) &&
-- 
1.6.3.3

